File: languages/go/entity/identity/multichoice_attribute.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

type MultichoiceAttribute struct {
	Attribute
}

func NewMultichoiceAttribute(key string, value string) MultichoiceAttribute {
	a := Attribute{
		Id:    key,
		Value: value,
	}
	return MultichoiceAttribute{
		Attribute: a,
	}
}

func NewMultichoiceAttributeFromProto(s *proto.MultiChoiceAttribute) MultichoiceAttribute {
	if s == nil {
		return MultichoiceAttribute{}
	}
	a := Attribute{
		Id:    s.GetId(),
		Value: s.GetValue(),
	}
	return MultichoiceAttribute{
		Attribute: a,
	}
}

func (s MultichoiceAttribute) ToProto() *proto.MultiChoiceAttribute {
	return &proto.MultiChoiceAttribute{
		Id:    s.Attribute.Id,
		Value: s.Attribute.Value.(string),
	}
}

File: languages/js/src/entity/identity/multichoice_attribute.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities";
import { Attribute } from "./attribute";

export class MultichoiceAttribute extends Attribute<string> {
  public toProto(): identityEntitiesProto.MultiChoiceAttribute {
    return identityEntitiesProto.MultiChoiceAttribute.fromPartial({
      id: this.id,
      value: this.value
    });
  }

  static fromProto(
    r: identityEntitiesProto.MultiChoiceAttribute
  ): MultichoiceAttribute {
    return new MultichoiceAttribute(r.id, r.value);
  }
}
